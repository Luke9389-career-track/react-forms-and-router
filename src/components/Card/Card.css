.Card{
  background: white;  
  border-radius: 10px;
  padding: 15px 5px;
  margin: 55px 5px;
  box-shadow: black 10px 10px 10px;
}

.Card:hover{
  box-shadow: rgb(79, 202, 244) 0px 0px 60px;
  border: rgb(200, 241, 255) solid 5px;
  animation: moveUp 2s infinite;
}

@keyframes moveUp {
    0% {transform: translate3d(0px,0px,0px);}
    50% {  transform: translate3d(0px,-15px,5px);} 
    100% {transform: translate3d(0px,0px,0px);}

}
