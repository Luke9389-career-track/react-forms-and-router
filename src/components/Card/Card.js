import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Card({ title, content, image }) {

  return (
    <Link>
      <div>
        {title && <header><h3>{title}</h3></header>}
        <figure>
          {image && <img src={image} alt={title ? title : null} />}
        </figure>
        {content && <p>{content}</p>}
      </div>
    </Link>

  );
}

Card.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  image: PropTypes.string,
};

export default Card;
