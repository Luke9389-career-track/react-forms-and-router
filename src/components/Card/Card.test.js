import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card component', () => {
  it('renders Card', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper).toMatchSnapshot();
  });
});
