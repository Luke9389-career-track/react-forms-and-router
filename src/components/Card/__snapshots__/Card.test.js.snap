// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders Card 1`] = `
<div>
  <figure />
</div>
`;
